compiler/src/org.graalvm.compiler.core.amd64/src/org/graalvm/compiler/core/amd64/AMD64ArithmeticLIRGenerator.java

```diff
@@ -961,6 +964,18 @@ public Variable emitBitScanReverse(Value value) {
         return result;
     }
 
+    @Override
+    public Variable emitFusedMultiplyAdd(Value a, Value b, Value c) {
+        Variable result = getLIRGen().newVariable(LIRKind.combine(a, b, c).changeType(AMD64Kind.DWORD));
```

Comment:
You don't need to change the result type to `DWORD`. 
```
Variable result = getLIRGen().newVariable(LIRKind.combine(a, b, c));
```
would be sufficient. If you want to be robust, try assert that a, b, c are of the same platform kind and are either `FLOAT` or `DOUBLE`.